app/src/main/java/com/example/teamproject/MapFragment.java

```diff
@@ -23,6 +25,16 @@
     MapView mMapView;
     View mView;
 
+    LocationResult locationResult = new LocationResult(){
```

Comment:
I would instantiate and store this object of type `LocationResult` within the `MyLocation` class. No need to expose this to the client `MapFragment` -- sometimes implementation details can and should be opaque. Exposing a `getLocation` method that retrieves the value from the `MyLocation` object can hide away all the setup and maintenance.

```
class MyLocation {

 private LocationResult locationResult;

 constructor() {
  this.locationResult = new LocationResult();
 }

 LocationResult getLocationResult() {
   return this.locationResult;
 }

}
```